server/handlers/auth.ts

```diff
@@ -111,18 +111,58 @@ export const admin: Handler = async (req, res, next) => {
   throw new CustomError("Unauthorized", 401);
 };
 
+export const googleLogin: Handler = async (req, res, next) => {
+  const salt = await bcrypt.genSalt(12);
+  const password = await bcrypt.hash(req.body.password, salt);
+  const user = await query.user.find({ email: req.body.email });
+
+  if (user) {
+    if (!user?.verified) {
+      if (env.DISALLOW_VERIFICATION) {
+        await query.user.update({ email: req.body.email }, { verified: true })
+      } else {
+        await mail.verification(user);
+        return res
+          .status(201)
+          .send({ message: "Verification email has been sent." });
+      }
+    }
+  } else {
+    if (env.DISALLOW_VERIFICATION) {
+      await query.user.add({ email: req.body.email, password, verified: true }, req.user);
+    } else {
+      await query.user.add({ email: req.body.email, password, verified: false }, req.user);
+      await mail.verification(user);
+      return res
+        .status(201)
+        .send({ message: "Verification email has been sent." });
+    }
+  }
```

Comment:
For the same reason as when the user has not yet been created (creating it in this case).
```suggestion
  } else {
    await query.user.add({ email: req.body.email, password, verified: true }, req.user);
  }
```